src/applications/vre/definitions/profileAddress.js

```diff
@@ -0,0 +1,265 @@
+      <span>
+        The United States is automatically chosen as your country if you live on
+        a military base outside of the country.
+      </span>
+    </AdditionalInfo>
+  </div>
+);
+
+/**
+ * @param {string} path - path to the address in formData, may contain [INDEX] as part of it, which needs to be handled using insertArrayIndex
+ * @param {string} checkBoxTitle - Visual label for the military base checkbox. Ex: "I live on a United States military base outside of the U.S."
+ * @param {function} callback - slots into ui:required for the necessary fields
+ *
+ * Conventions:
+ * 1. formDataPath - path to entire address property in formData, after accounting for potential array nesting. Derived from the path parameter
+ * 2. get(formDataPath, formData) - returns the address property, often destructured with const {country, isMilitary} = get(formDataPath, formData)
+ *
+ * Examples:
+ * 1. Path to Address nested in array - childrenToAdd[INDEX].childAddressInfo.address
+ */
+
+export const addressUiSchema = (path, checkBoxTitle, callback) => {
+  // Used when addresses are nested in an array and need to be assessible.
+  const insertArrayIndex = (key, index) => key.replace('[INDEX]', `[${index}]`);
+  // the address is nested in an array, handle index
+  const getPath = (pathToData, index) =>
+    typeof index === 'number'
+      ? insertArrayIndex(pathToData, index)
+      : pathToData;
+
+  return (function returnAddressUI() {
+    return {
+      isMilitary: {
+        'ui:title': checkBoxTitle,
+      },
+      'view:livesOnMilitaryBaseInfo': {
+        'ui:description': MilitaryBaseInfo,
+      },
+      country: {
+        'ui:required': callback,
+        'ui:title': 'Country',
+        'ui:options': {
+          updateSchema: (formData, schema, uiSchema, index) => {
+            const formDataPath = getPath(path, index);
+            const countryUI = uiSchema;
+            const addressFormData = get(formDataPath, formData);
+            const { isMilitary } = addressFormData;
+            // if isMilitary === true, auto select United States and disable the field
+            if (isMilitary) {
+              countryUI['ui:disabled'] = true;
+              addressFormData.country = USA.value;
+              return {
+                enum: [USA.value],
+                enumNames: [USA.label],
+                default: USA.value,
+              };
+            }
+            // default to regular country select dropdown
+            countryUI['ui:disabled'] = false;
+            return {
+              type: 'string',
+              enum: countries.map(country => country.value),
+              enumNames: countries.map(country => country.label),
+            };
+          },
+        },
+      },
+      addressLine1: {
+        'ui:required': callback,
+        'ui:title': 'Street',
+        'ui:errorMessages': {
+          required: 'Street address is required',
+          pattern: 'Street address must be under 100 characters',
```

Comment:
:thinking: Does this show for the `maxLength` validation error? I thought this was only for the `pattern` violation (which in this case would be the `/^.*\\S.*/` regex).

Reply:
Also, I the messaging guidelines say ([ref](https://github.com/department-of-veterans-affairs/vets.gov-team/blob/master/Products/Platform/Design%20System/Guidelines/Error%20handling/Dictionary.md#task-completion-successfailure)):

> Please fill in a valid [address/email address/phone number].

Reply:
👀  had no idea that messaging guideline existed!! I've been winging a lot of the error messages =\ not a good look on my part. 

Good thought, Chris. I'll go back and take a look at this.

Reply:
Update: @cvalarida is right. This pattern error message maps to the regex used, not `maxLength`.